File: java/parser/src/main/java/gov/lbl/parser/parser/Named_argumentVisitor.java
package gov.lbl.parser.parser;

import gov.lbl.parser.domain.Function_argument;
import gov.lbl.parser.domain.Named_argument;
import gov.lbl.antlr4.visitor.modelicaBaseVisitor;
import gov.lbl.antlr4.visitor.modelicaParser;

public class Named_argumentVisitor extends modelicaBaseVisitor<Named_argument> {
    @Override
    public Named_argument visitNamed_argument(modelicaParser.Named_argumentContext ctx) {
        String identifier = ctx.IDENT() == null? "" : ctx.IDENT().getText();
        Function_argumentVisitor function_argumentVisitor = new Function_argumentVisitor();
        Function_argument value = ctx.function_argument().accept(function_argumentVisitor);
        return new Named_argument(identifier, value);
    }
}

File: jsParser/parser/Named_argumentVisitor.js
const modelicaVisitor = require('../antlrFiles/modelicaVisitor').modelicaVisitor
const Named_argument = require('../domain/Named_argument')

const Function_argumentVisitor = require('./Function_argumentVisitor')

class Named_argumentVisitor {
  constructor () {
    modelicaVisitor.call(this)
    return this
  }

  visitNamed_argument (ctx) {
    const identifier = ctx.IDENT() ? ctx.IDENT().getText() : ''
    let value = null

    if (ctx.function_argument()) {
      const function_argumentVisitor = new Function_argumentVisitor.Function_argumentVisitor()
      value = function_argumentVisitor.visitFunction_argument(ctx.function_argument())
    }

    return new Named_argument.Named_argument(identifier, value)
  }
};

Named_argumentVisitor.prototype = Object.create(modelicaVisitor.prototype)

exports.visitNamed_argument = this.visitNamed_argument
exports.Named_argumentVisitor = Named_argumentVisitor
